### DISCUSSION TITLE
Recomended maximum limit of products

### DISCUSSION THREAD

**Discussion by welectronic** [medusajs/medusa#5443] [Q&A] [answered]
in terms of performance, What is the recomended maximum limit of products, variants, or SKUs? I am conducting a benchmark for a B2B and B2C commerce platform with over 30,000 SKUs per country, in different languages, across various countries, and with distinct warehouses and locations

**Comment by olivermrbl** [comment, marked as answer]
We are yet to test a potential maximum limit of products, so, unfortunately, I can't recommend anything to you. I will say that we have worked with several companies running stores with millions of variants without facing performance issues, so if there is a limit, it is still to be hit by them. If this happens, I'll make sure to let you know. Hope this helps.
